msg.go

```diff
@@ -62,13 +65,24 @@ var (
 	ErrNoSig         error = &Error{err: "no signature found"}
 	ErrPrivKey       error = &Error{err: "bad private key"}
 	ErrRcode         error = &Error{err: "bad rcode"}
-	ErrRdata         error = &Error{err: "bad rdata"}
 	ErrRRset         error = &Error{err: "bad rrset"}
 	ErrSecret        error = &Error{err: "no secrets defined"}
 	ErrShortRead     error = &Error{err: "short read"}
 	ErrSig           error = &Error{err: "bad signature"} // ErrSig indicates that a signature can not be cryptographically validated.
 	ErrSoa           error = &Error{err: "no SOA"}        // ErrSOA indicates that no SOA RR was seen when doing zone transfers.
 	ErrTime          error = &Error{err: "bad time"}      // ErrTime indicates a timing error in TSIG authentication.
+
+	// TODO(tmthrgd): Stop returning and deprecate ErrRdata. It's not well
+	// defined.
+	ErrRdata error = &Error{err: "bad rdata"}
+
+	// For CoreDNS, we need to use an error that either matches
+	// errors.Is(_, ErrBuf) or contains the substring "overflow".
+	//
+	// TODO(tmthrgd): Update CoreDNS and use "truncated message".
+	errTruncatedMessage = &Error{err: "overflow unpacking truncated message"}
+	errUnpackOverflow   = &Error{err: "overflow unpacking data"}
```

Comment:
data -> rdata?

Reply:
Well this one gets returned for both msg header, question headers, resource headers _and_ resource data. I'm not sure we can be more specific here.